### PR TITLE
Add BIP 66 notice to the 0.10 release notes

### PR DESCRIPTION
**Pull request by sipa**

### CODE REVIEW

File: doc/release-notes.md [hunk 1]
```diff
@@ -293,6 +293,22 @@ if this is 1.
...
 
+BIP 66: strict DER encoding for signatures
+------------------------------------------
+
+Bitcoin Core 0.10 implements BIP 66, which introduces block version 3.
+
+When a sequence of 1001 blocks is produced out of which 751 have version 3 or
+higher set, a new consensus rule becomes active, which prohibits non-DER
+signatures. Such transactions have been non-standard since Bitcoin v0.8.0
+(released in Februari 2013), but were technically still permitted inside
```
Comment (sdaftuar) [inline doc/release-notes.md]: typo: Februari -> February

**Comment by sipa** [comment]
Rewritten a bit.

@sdaftuar Fixed.

File: doc/release-notes.md [hunk 2]
```diff
@@ -293,6 +293,24 @@ if this is 1.
...
+non-standard since Bitcoin v0.8.0 (released in February 2013), but were
+technically still permitted inside blocks.
+
+This change breaks the dependency on OpenSSL's signature parsing, and is
+required if implementations would want to remove all of OpenSSL from the
+consensus code.
+
+The same miner-voting mechanism as in BIP 34 is used: when 751 out of a
+sequence of 1001 blocks have version number 3 or higher, the new consensus
+rule becomes active for those blocks. When 951 out of sequence of 1001
```
Comment (sdaftuar) [inline doc/release-notes.md]: (sorry for the nits): "951 out of sequence" should be "951 out of a sequence"

rest looks good to me!

**Comment by sipa** [comment]
@sdaftuar Thanks for looking; fixed.

**Comment by theuni** [comment]
ACK.

File: doc/release-notes.md [hunk 3]
```diff
@@ -293,6 +293,24 @@ if this is 1.
...
+------------------------------------------
+
+Bitcoin Core 0.10 implements BIP 66, which introduces block version 3, and a new
+consensus rule, which prohibits non-DER signatures. Such transactions have been
+non-standard since Bitcoin v0.8.0 (released in February 2013), but were
+technically still permitted inside blocks.
+
+This change breaks the dependency on OpenSSL's signature parsing, and is
+required if implementations would want to remove all of OpenSSL from the
+consensus code.
```
Comment (luke-jr) [inline doc/release-notes.md]: I would mention that OpenSSL does not guarantee consensus-compatible behaviour here too.

"This change breaks the dependency on implementation-specific details in OpenSSL's signature parsing, avoids depending on OpenSSL maintaining those details to be consensus-compatible (which they do not guarantee), and is required if implementations would want to remove all of OpenSSL from the consensus code."

**Comment by luke-jr** [comment]
I suggest perhaps moving this section up before "REST interface" since it affects everyone. OTOH, it's not really user-visible, so maybe not. *shrug*

**Comment by fanquake** [comment]
ACK

File: doc/release-notes.md (hunk 3 above)
Comment (sipa) [inline doc/release-notes.md]: Meh.

**Comment by laanwj** [comment]
Merged via e43f25c (had to resolve a trivial merge conflict with #5755)

**Pull request closed without merging**